admin/app.js

```diff
@@ -210,19 +211,32 @@ passport.deserializeUser(function (user, done) {
   done(null, user)
 })
 
-// passport with custom strategy
-passport.use(new CustomStrategy(
-  require('./authentication/nca-tools-authentication-strategy')
-))
-
-// Passport with local strategy
-passport.use(
-  new LocalStrategy({
-    passReqToCallback: true
-  },
-  require('./authentication/local-strategy')
-  )
-)
+// initialise chosen auth strategy only
+switch (config.Auth.mode) {
+  case authModes.dfeSignIn:
+    ;(async function () {
```

Comment:
This IIFE is not synchronised with the switch statement at all.  Is there any way to remove the await on the `dfe-signin-strategy` and make this synchronous?

Reply:
yes, we could make the initialisation of OpenIdConnect synchronous, which is a fairly valid ask given that the app should not even start if DfE signin service is not available when enabled.

Reply:
i've added a synchronous implementation, but it fails sporadically on discovery, likely due to the signin service configuring short request timeouts.

Reply:
switched back to async, with promise based call in app.js